src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1387,7 +1387,7 @@ private static object[] ToObjectArray(IEnumerable<IThreadPoolWorkItem> workitems
             i = 0;
             foreach (IThreadPoolWorkItem item in workitems)
             {
-                if (i < result.Length) //just in case someone calls us while the queues are in motion
+                if ((uint)i < (uint)result.Length) //just in case someone calls us while the queues are in motion
```

Comment:
It also won't matter in this particular case, as this code is not particularly performance-sensitive: it's only called from a debugger when it asks for the list of queued work items to show to the developer.